drivers/led_strip/ws2812_spi.c

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ * Copyright (c) 2019, Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/led_strip.h>
+
+#include <string.h>
+
+#define LOG_LEVEL CONFIG_LED_STRIP_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(ws2812_spi);
+
+#include <zephyr.h>
+#include <device.h>
+#include <drivers/spi.h>
+#include <sys/math_extras.h>
+#include <sys/util.h>
+
+/*
+ * SPI master configuration:
+ *
+ * - mode 0 (the default), 8 bit, MSB first (arbitrary), one-line SPI
+ * - no shenanigans (don't hold CS, don't hold the device lock, this
+ *   isn't an EEPROM)
+ */
+#define SPI_OPER (SPI_OP_MODE_MASTER | SPI_TRANSFER_MSB | \
+		  SPI_WORD_SET(8) | SPI_LINES_SINGLE)
```

Comment:
I'll add a couple of macros for magic numbers, but the value 8 appears sometimes because a struct led_rgb color channel is defined as a u8_t, so I'm going to leave those (such as in ws2812_spi_ser()) as-is.